app/src/main/java/com/github/se/eduverse/model/folder/FolderViewModel.kt

```diff
@@ -0,0 +1,95 @@
+package com.github.se.eduverse.model.folder
+
+import android.util.Log
+import androidx.lifecycle.ViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+
+class FolderViewModel(val repository: FolderRepository) : ViewModel() {
+
+  private val _existingFolders: MutableStateFlow<MutableList<Folder>> =
```

Comment:
I think that it should be initialized as an emptyList. And it's better to name it _folders. Also I think it should be a MutableStateFlow of  a list not mutable list for the same reasons explained in the Folder.ket comment

Reply:
The problem with list is that you can't add or remove object from them, so by changing it I would need to recreate a list everytime something change in it, which is not very efficient.
I changed the initialisation and the name.

Reply:
In fact you don't need to manage the value(of type list) held by the mutable state flow, each time some action is performed on the folder collection of the db you need to call the getFolders and on success it just assigns the result list to the value of the mutable state flow